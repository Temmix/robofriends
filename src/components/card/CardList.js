import React from 'react';
import Card from './Card';

const CardList = ({robots}) => {  
    return (
        <div>
          {
              robots.map((x,i) => 
                <Card 
                    key={i} 
                    id={x.id} 
                    email={x.email} 
                    name={x.name}/>
                )
          }
         </div>
        );
}

export default CardList;