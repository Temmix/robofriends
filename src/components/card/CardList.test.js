import { shallow, mount, render } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('expect to render CardList component', () => {
    const mockedRobots = [
        {
            id: 1,
            name: 'Tim Tolly',
            email: 'Tim.Tolly@gmail.com'
        },
        {
            id: 2,
            name: 'Franc Jessica',
            email: 'Franc.Jessica@gmail.com'
        }

    ]
    expect(shallow(<CardList robots={mockedRobots}/>)).toMatchSnapshot();
})