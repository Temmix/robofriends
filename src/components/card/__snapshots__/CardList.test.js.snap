// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
      Array [
        Object {
          "email": "Tim.Tolly@gmail.com",
          "id": 1,
          "name": "Tim Tolly",
        },
        Object {
          "email": "Franc.Jessica@gmail.com",
          "id": 2,
          "name": "Franc Jessica",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="Tim.Tolly@gmail.com"
          id={1}
          name="Tim Tolly"
        />,
        <Card
          email="Franc.Jessica@gmail.com"
          id={2}
          name="Franc Jessica"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "email": "Tim.Tolly@gmail.com",
          "id": 1,
          "name": "Tim Tolly",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "Franc.Jessica@gmail.com",
          "id": 2,
          "name": "Franc Jessica",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="Tim.Tolly@gmail.com"
            id={1}
            name="Tim Tolly"
          />,
          <Card
            email="Franc.Jessica@gmail.com"
            id={2}
            name="Franc Jessica"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "email": "Tim.Tolly@gmail.com",
            "id": 1,
            "name": "Tim Tolly",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "Franc.Jessica@gmail.com",
            "id": 2,
            "name": "Franc Jessica",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
